### ISSUE TITLE
MaxPool padding settings Pytorch->ONNX->Tensorflow

### ISSUE THREAD

**Issue by Dnekhaev** [onnx/onnx#388]
I try to convert nn.MaxPool2d to onnx as

dummy_input = Variable(torch.randn(1, 1, 4, 4))
model = nn.MaxPool2d(kernel_size=3,stride=2, padding=1)
torch.onnx.export(model, dummy_input, "./maxpool.pb", verbose=True)

I get output:

graph(%1 : Float(1, 1, 4, 4)) {
  %2 : Float(1, 1, 2, 2) = MaxPool[kernel_shape=[3, 3], pads=[1, 1], strides=[2, 2]](%1), uses = [%0.i0];
  return (%2);
}

Then using tensorflow backend load the model

model = onnx.load('./maxpool.pb')
tf_rep = prepare(model)

For input np.arange(16).reshape(1,1,4,4) in pytorch I have output [[5,7],[13,15]] and in tensorflow the output is [[10,11],[14,15]]. How can I properly set padding settings while converting to onnx?

**Comment by dzhulgakov** [comment]
It looks like the famous case of TF implementing SAME padding being weighted to the right. If you look at the outputs it basically means that TF interprets the pads as [0,0,1,1]. See discussion in https://github.com/onnx/onnx/issues/185#issuecomment-342359888 for some context.

@tjingrant - seem like an issue in onnx-tf, can somebody help to take a look?

**Comment by ezyang** [comment]
@Dnekhaev Your problem might be solved by https://github.com/onnx/onnx-tensorflow/pull/37 ; give it a try.
